common/src/main/java/com/wynntils/models/items/encoding/impl/block/PowderDataTransformer.java

```diff
@@ -4,17 +4,76 @@
  */
 package com.wynntils.models.items.encoding.impl.block;
 
+import com.wynntils.models.elements.type.Powder;
 import com.wynntils.models.items.encoding.data.PowderData;
 import com.wynntils.models.items.encoding.type.DataTransformer;
 import com.wynntils.models.items.encoding.type.ItemTransformingVersion;
+import com.wynntils.utils.UnsignedByteUtils;
 import com.wynntils.utils.type.ErrorOr;
+import com.wynntils.utils.type.Pair;
 import com.wynntils.utils.type.UnsignedByte;
+import java.util.stream.Stream;
+
+public class PowderDataTransformer extends DataTransformer<PowderData> {
+    public static final byte ID = 4;
 
-public class PowderDataTransformer implements DataTransformer<PowderData> {
     @Override
     public ErrorOr<UnsignedByte[]> encodeData(ItemTransformingVersion version, PowderData data) {
         return switch (version) {
-            case VERSION_1 -> ErrorOr.of(new UnsignedByte[0]);
+            case VERSION_1 -> encodePowderData(data);
         };
     }
+
+    @Override
+    protected boolean shouldEncodeData(ItemTransformingVersion version, PowderData data) {
+        return !data.powders().isEmpty();
+    }
+
+    @Override
+    protected byte getId() {
+        return ID;
+    }
+
+    private ErrorOr<UnsignedByte[]> encodePowderData(PowderData data) {
+        // Powders are encoded as bits, a powder needs 5 bits to encode
```

Comment:
So if we ever get a way to find the tier of applied powders, we will solve this by retiring this transformer, and creating a new with another ID and using that instead? Is that the general thinking?

Reply:
No, it supports powder tier encoding. So we just assume every powder is tier 6. Although yes, the general thinking would be to up the version, if some unsolvable conflict happens.

Reply:
I think I'm missing where the powder tier is encoded then..? Can you point me to it?